### PR TITLE
Clarify RP conventions with Razor components

### LINKED ISSUE #16798: Is `AddRazorPagesOptions` fully supported for Blazor?
The documentation does not mention whether the method `services.AddRazorPages().AddRazorPagesOptions(options => { });` is supported in Blazor. If it is,
an example should probably be provided (calling out any differences) as well as a link to https://docs.microsoft.com/en-us/aspnet/core/security/authorization/razor-pages-authorization for more information.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c9942b0a-a7fb-0d34-9201-af0963affa9e
* Version Independent ID: 19770710-5225-a1ea-e082-65da6ba4a49a
* Content: [ASP.NET Core Blazor authentication and authorization](https://docs.microsoft.com/en-us/aspnet/core/security/blazor/?view=aspnetcore-3.1&tabs=visual-studio#authorization)
* Content Source: [aspnetcore/security/blazor/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/security/blazor/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #16798

I think this is correct, but I haven't tested the first scenario (routeable components) myself. The second scenario (embedded) is obviously true.

### CODE REVIEW

**Comment by guardrex** [comment]
@Rick-Anderson Let's do this one, too. It's very straightforward ... just saying that components are only *indirectly effected* by RP conventions when the component is embedded in a page.

File: aspnetcore/security/blazor/index.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ Security scenarios differ between Blazor Server and Blazor WebAssembly apps. Bec
 
 Blazor WebAssembly apps run on the client. Authorization is *only* used to determine which UI options to show. Since client-side checks can be modified or bypassed by a user, a Blazor WebAssembly app can't enforce authorization access rules.
 
+[Razor Pages authorization conventions](xref:security/authorization/razor-pages-authorization) don't apply to routable Razor components. If an unroutable Razor component is [embedded in a page](xref:blazor/integrate-components#render-components-from-a-page-or-view), the Razor component is indirectly effected by the page's authorization conventions along with the rest of the page's content.
```
Comment (mkArtakMSFT) [inline aspnetcore/security/blazor/index.md]: `effected` or `affected`?

Comment (guardrex) [reply aspnetcore/security/blazor/index.md]: *Yes* ... "affected" for **_impact on_**. Good :eye:!

File: aspnetcore/security/blazor/index.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ Security scenarios differ between Blazor Server and Blazor WebAssembly apps. Bec
 
 Blazor WebAssembly apps run on the client. Authorization is *only* used to determine which UI options to show. Since client-side checks can be modified or bypassed by a user, a Blazor WebAssembly app can't enforce authorization access rules.
 
+[Razor Pages authorization conventions](xref:security/authorization/razor-pages-authorization) don't apply to routable Razor components. If an unroutable Razor component is [embedded in a page](xref:blazor/integrate-components#render-components-from-a-page-or-view), the Razor component is indirectly affected by the page's authorization conventions along with the rest of the page's content.
```
Comment (mkArtakMSFT) [inline aspnetcore/security/blazor/index.md]: What about this minor rewording?
```suggestion
[Razor Pages authorization conventions](xref:security/authorization/razor-pages-authorization) don't apply to routable Razor components. If a non-routable Razor component is [embedded in a page](xref:blazor/integrate-components#render-components-from-a-page-or-view), the Razor component is indirectly affected by the page's authorization conventions along with the rest of the page's content.
```

**Comment by guardrex** [comment]
I tested the routable scenario with a page route convention and a routable Razor component, and the test result reflects the concept described. Of course, an embedded non-routable component is like anything else on a page controlled by a convention.

**Pull request merged**